genomeGraph/giraf.go

```diff
@@ -115,7 +118,9 @@ func RandSomaticMutations(graph *GenomeGraph, reads []*giraf.Giraf, numSomaticSN
 						continue
 					}
 					readPos = NodePosToReadPos(graph, reads[j], nodeIdx, pos)
-
+					if int(readPos) >= len(reads[j].Seq) {
```

Comment:
What is this checking?  Is this fixing a bug?

Reply:
Fixing a bug. The RandLocationFast on newline 107 does not guarentee to pull a pos within the read, so I was getting index out of range errors. This check ensures the pos is valid.